aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIO.java

```diff
@@ -453,7 +453,7 @@ public void initialize(Map<String, String> props) {
 
     // Do not override s3 client if it was provided
     if (s3 == null) {
-      Object clientFactory = S3FileIOAwsClientFactories.initialize(props);
+      Object clientFactory = S3FileIOAwsClientFactories.initialize(propertiesWithCredentials);
```

Comment:
FYI this is going to be fixed by https://github.com/apache/iceberg/pull/12799 but it's still good to get the tests in

Reply:
> FYI this is going to be fixed by #12799 but it's still good to get the tests in
Thanks for the reminder, this feature looks great!